test/e2e/scheduling/priorities.go

```diff
@@ -425,7 +430,23 @@ var _ = SIGDescribe("SchedulerPriorities [Serial]", func() {
 })
 
 // createBalancedPodForNodes creates a pod per node that asks for enough resources to make all nodes have the same mem/cpu usage ratio.
-func createBalancedPodForNodes(f *framework.Framework, cs clientset.Interface, ns string, nodes []v1.Node, requestedResource *v1.ResourceRequirements, ratio float64) error {
+func createBalancedPodForNodes(f *framework.Framework, cs clientset.Interface, ns string, nodes []v1.Node, requestedResource *v1.ResourceRequirements, ratio float64) (func(), error) {
+	cleanUp := func() {
+		// Delete all remaining pods
+		podList, err := cs.CoreV1().Pods(ns).List(context.TODO(), metav1.ListOptions{
+			LabelSelector: labels.SelectorFromSet(labels.Set(balancePodLabel)).String(),
+		})
+		if err != nil {
+			framework.Logf("Failed to list remaining pods for cleanup: %v.", err)
+		} else {
+			for _, pod := range podList.Items {
+				if err := e2epod.DeletePodWithWaitByName(cs, pod.Name, pod.Namespace); err != nil {
```

Comment:
can we parallelize and have a wait group?
Alternatively, you can do `DeleteCollection` and have a `wait.Poll` that does a `List` until it returns empty.

Reply:
Done